napari-workshops/notebooks/spot_detection_basic.md

```diff
@@ -0,0 +1,270 @@
+
+## Load the data
+
+In the cells below load the data using the scikit-image `imread()` function. For
+more information about the `imread()` function, please see the [scikit-image docs](https://scikit-image.org/docs/dev/api/skimage.io.html#skimage.io.imread). We are loading two images:
+
+- `nuclei`: an image of cell nuclei
+- `spots`: an image of in situ sequencing spots
+
+```{code-cell} ipython3
+from skimage import io
+
+nuclei_url = 'https://raw.githubusercontent.com/kevinyamauchi/napari-spot-detection-tutorial/main/data/nuclei_cropped.tif'
+nuclei = io.imread(nuclei_url)
+
+spots_url = 'https://raw.githubusercontent.com/kevinyamauchi/napari-spot-detection-tutorial/main/data/spots_cropped.tif'
+spots = io.imread(spots_url)
+```
+
+## View the data
+
+To view our data, we will create a napari viewer and then we will add the images to the viewer
+via the viewer's `add_image()` method. We will also import the `nbscreenshot` utility.
+
+```{code-cell} ipython3
+import napari
+from napari.utils import nbscreenshot
+
+# create the napari viewer
+viewer = napari.Viewer()
+
+# add the nuclei image to the viewer
+viewer.add_image(nuclei, colormap='green')
+
+# add the spots image to the viewer
+viewer.add_image(spots, colormap='magenta', blending='additive')
+```
+
+After loading the data, inspect it in the viewer and adjust the
+layer settings to your liking (e.g., contrast limits, colormap). 
+You can pan/zoom around the image by click/dragging to pan and scrolling with your
+mousewheel or trackpad to zoom.
+
+```{tip}
+You can adjust a layer's opacity to see the change how much you see of the
+layers that are "under" it.
+```
+For example, for better contrast you could choose an inverted colormap and minimum blending.
+
+```{code-cell} ipython3
+viewer.layers['nuclei'].colormap = 'I Forest'
+viewer.layers['nuclei'].blending = 'minimum'
+viewer.layers['spots'].colormap = 'I Orange'
+viewer.layers['spots'].blending = 'minimum'
+```
+
+Once you are satisfied, you can print the output of any manual changes and then take a screenshot of the viewer.
+
+```{code-cell} ipython3
+# example of printing the nuclei layer visualization params
+print('Colormap: ', viewer.layers['nuclei'].colormap.name)
+print('Contrast limits: ', viewer.layers['nuclei'].contrast_limits)
+print('Opacity: ', viewer.layers['nuclei'].opacity)
+print('Blending: ', viewer.layers['nuclei'].blending)
+```
+
+```{code-cell} ipython3
+nbscreenshot(viewer)
+```
+
+## Create an image filter
+
+If you look carefull at the `spots` layer, you will notice that it contains background and
+autofluorescence from the cells. It may help to just look at the single channel.
+
+```{code-cell} ipython3
+viewer.layers['nuclei'].visible = False
+```
+
+To improve spot detection, we will apply a high pass filter to improve the contrast of the spots.
+
+A simple way to achieve this is to:
+1. blur the image with a Gaussian, which removes high-frequency information—this is why we use it for denoising.
```

Comment:
just fyi, I only understood the meaning of "high pass filter" and "high frequency information" very late in my image analysis journey. So it may be useful to define it, or to use simpler terms like, "which blurs out small features — including our spots — and leaves intact only the larger features, including background intensity"